source/core/TypeAndDataManager.cpp

```diff
@@ -852,6 +852,8 @@ Boolean TypeCommon::IsA(const SubString *otherTypeName)
         if (t->Name().Compare(otherTypeName)) {  // ??? should this really consider unnamed types equal?? -CS
             return true;
         }
+        if (t->BitEncoding() == kEncoding_RefNum)
```

Comment:
Should we do this check instead similar to checking thisEncoding and otherEncoding for  kEncoding_Array and kEncoding_Cluster in the two functions below?

Boolean TypeCommon::IsA(TypeRef otherType, Boolean compatibleStructure)
Boolean TypeCommon::CompareType(TypeRef otherType)

This particular IsA() seems very generic and polluting this with specific encoded type does not look good to me.

Reply:
That other code is checking for intrinsics, and doesn't have  to deal with type inheritance.
I understand your concern, but unfortnately this check has to be here because this is the only place that it iterating through the type inheritance chain to look for a common base type, and that is the source of the problem.
I think we could improve this by adding a type predicate like "IsOpaqueReferencingType()" that could return true for refnums and possibly others such PointerType, and change this explicit check to use that.

Reply:
Oh, and yes, isFlat = true is correct.  The data is an integer refnum value and the max queue size.

Reply:
Please add "IsOpaqueReferencingType()" and leave a TODO to check other types for PointerType. If possible, I would recommend adding tests that may fail now and keep them disabled too -- while this is fresh in the mind.

Reply:
Added this.
PointerType is internal and doesn't exist in via files so it's not possible to write a test for this, but I think because of this we never call IsA on PointerType either so it should not matter.  Mentioned this in the TODO.